src/main/scala/com/advancedtelematic/campaigner/http/CampaignResource.scala

```diff
@@ -44,6 +47,32 @@ class CampaignResource(extractAuth: Directive1[AuthedNamespaceScope],
   def retryFailedDevices(ns: Namespace, mainCampaign: Campaign, request: RetryFailedDevices): Future[CampaignId] =
     campaigns.retryCampaign(ns, mainCampaign, request.failureCode)
 
+
+  implicit val installationFailureCsvMarshaller: ToResponseMarshaller[Seq[(String, String, String)]] =
+    Marshaller.withFixedContentType(ContentTypes.`text/csv(UTF-8)`) { t =>
+      val csv = CsvSerializer.asCsv(Seq("Device ID", "Failure Code", "Failure Description"), t)
+      val e = HttpEntity(ContentTypes.`text/csv(UTF-8)`, csv)
+      val h = `Content-Disposition`(ContentDispositionTypes.attachment, Map("filename" -> "device-failures.csv"))
```

Comment:
Any way to put `campaignId` into filename?

Reply:
I found a way although it's not very elegant. See my last commit.